modules/gateway/stream.go

```diff
@@ -0,0 +1,79 @@
+package gateway
+
+import (
+	"net"
+
+	"github.com/NebulousLabs/Sia/build"
+	"github.com/NebulousLabs/muxado"
+	"github.com/xtaci/smux"
```

Comment:
not sure whether we want to fork this (like muxado) or vendor it (like twofish). I think vendoring is considered best practice now, and a lot of tooling is being written for it, so my inclination is to vendor.